bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayTriggerHandler.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2017 Kai Kreuzer and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.automation.module.timer.handler;
+
+import org.eclipse.smarthome.automation.Trigger;
+import org.eclipse.smarthome.automation.handler.BaseTriggerModuleHandler;
+import org.eclipse.smarthome.automation.handler.RuleEngineCallback;
+import org.eclipse.smarthome.automation.module.timer.factory.TimerModuleHandlerFactory;
+import org.eclipse.smarthome.core.scheduler.Expression;
+import org.eclipse.smarthome.core.scheduler.ExpressionThreadPoolManager;
+import org.eclipse.smarthome.core.scheduler.ExpressionThreadPoolManager.ExpressionThreadPoolExecutor;
+import org.eclipse.smarthome.core.scheduler.RecurrenceExpression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an ModuleHandler implementation for Triggers which trigger the rule
+ * based on a cron expression. The cron expression can be set with the
+ * configuration.
+ *
+ * @author Kai Kreuzer - Initial Contribution
+ *
+ */
+public class TimeOfDayTriggerHandler extends BaseTriggerModuleHandler implements Runnable {
+
+    private final Logger logger = LoggerFactory.getLogger(TimeOfDayTriggerHandler.class);
+
+    public static final String MODULE_TYPE_ID = "timer.TimeOfDayTrigger";
+    public static final String MODULE_CONTEXT_NAME = "MODULE";
+
+    private static final String CFG_TIME = "time";
+
+    private final ExpressionThreadPoolExecutor scheduler;
+    private final Expression expression;
+
+    public TimeOfDayTriggerHandler(Trigger module) {
+        super(module);
+        String time = (String) module.getConfiguration().get(CFG_TIME);
```

Comment:
Above you suppress warning about unchecked casts.
Isn't this necessary here, too?
Okay, it is only a warning, so it is not really necessary, but if used above, why not using here?